comp/otelcol/otlp/components/connector/datadogconnector/connector.go

```diff
@@ -0,0 +1,214 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogconnector // import "github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/connector/datadogconnector"
+
+import (
+	"context"
+	"fmt"
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog"
+	datadogconfig "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog/config"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/statsprocessor"
+	"github.com/DataDog/datadog-agent/pkg/obfuscate"
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/metric/noop"
+	"go.uber.org/zap"
+)
+
+// traceToMetricConnector is the schema for connector
+type traceToMetricConnector struct {
+	metricsConsumer consumer.Metrics // the next component in the pipeline to ingest metrics after connector
+	logger          *zap.Logger
+
+	// concentrator ingests spans and produces APM stats
+	concentrator *stats.Concentrator
+
+	// tcfg is the trace agent config
+	tcfg *config.AgentConfig
```

Comment:
OK to leave these as-is for now but please leave a TODO to use concentrator etc. from the trace-agent component. Right now you are creating (unnecessary) multiple concentrator, obfuscator, agent configs etc. in otel-agent